### PR TITLE
Update the hr-time IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

File: interfaces/hr-time.idl [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 [Exposed=(Window,Worker)]
 interface Performance : EventTarget {
-    DOMHighResTimeStamp now ();
+    DOMHighResTimeStamp now();
     readonly attribute DOMHighResTimeStamp timeOrigin;
     [Default] object toJSON();
 };
-
-partial interface WindowOrWorkerGlobalScope {
+partial interface mixin WindowOrWorkerGlobalScope {
```
Comment (foolip) [inline interfaces/hr-time.idl]: Typo in spec here? Doesn't this parse error break the test entirely?

Comment (lukebjerring) [reply interfaces/hr-time.idl]: What's supposed to be a typo?

**Comment by lukebjerring** [comment]
Tweaked the test to cover `Window` and `WorkerGlobalScope` until https://github.com/web-platform-tests/wpt/issues/8053 is fixed. PTAL.

**Review by foolip — APPROVED** [review]
Tested locally in Chrome and Firefox, 20/20 pass in both.

**Pull request merged**